File: arraywithoddsum.java
import java.util.Scanner;

public class arraywithoddsum{

	public static void main(String[] args) {
		
		Scanner in = new Scanner(System.in);

		int t = in.nextInt();

		while (t-- > 0) {
			
			int cnt = 0;
			int even = 0;
			int odd = 0;
			int sum = 0;

			int n = in.nextInt();

			for (int i = 0; i < n; i++) {

				cnt = in.nextInt();
				sum+=cnt;

				if (cnt % 2 == 0) {

					even++;
				
				}

				else {

					odd++;
				
				}
			}

			if((sum % 2 != 0) || (even >= 1 && odd >=1)){
				System.out.println("YES");
			
			}

			else{

				System.out.println("NO");
			
			}
		}
	}
}

File: arraywithoddsum.py
t = int(input())
for i in range(t):
    odd = 0
    even = 0
    tot = 0
    input()
    a = list(map(int, input().split()))
    for i in a:
        if i % 2 != 0:
            odd+=1
        else:
            even+=1
        tot+=i
    if (tot % 2 != 0) or (odd >= 1 and even >= 1):
        print("YES")
    else:
        print("NO")



